### DISCUSSION TITLE
Poco 1.10.1 Visual Studio 19 OpenSSL Compiling Question

### DISCUSSION THREAD

**Discussion by Muppet1856** [pocoproject/poco#3179] [Q&A]
I've setup my buildwin.ps1 correctly to compile with OpenSSL (I think) but it keeps failing when it gets to crypto.  I downloaded the master branch from GIT, and had to update the `<WindowsTargetPlatformVersion>` to 10 in CPPUnit and Foundation (VS19 doesn't come with SDK 8.1).  

I kick it all off in PowerShell with:
`.\buildwin.ps1 -poco_base . -vs 160 -action build -linkmode all -config both -platform Win32 -samples -tests -verbosity m -openssl_base C:\OpenSSL\Win32 -mysql_base C:\MySQL\release`

One thing that I just noticed is this too long line:

`
The input line is too long.
The syntax of the command is incorrect.
Build configuration:
--------------------
Poco Base:     .
Version:       160
Action:        build
Link Mode:     all
Configuration: both
Platform:      Win32
Tests:         True
Samples:       True
Build Tool:    msbuild
OpenSSL:       c:\OpenSSL\Win32
MySQL:         C:\MySQL\release`

Everything is humming along until I get to Crypto, then:
`C:\Poco\Crypto\include\Poco\Crypto\Crypto.h(28,10): fatal error C1083: Cannot open include file: 'openssl/opensslv.h': No such file or directory (compiling source file sr
c\CryptoTransform.cpp) [C:\Poco\Crypto\Crypto_vs160.vcxproj]`

When I check the environment variables, 
`Get-ChildItem -Path Env:INCLUDE|format-list` 

I get 
`Name  : INCLUDE
Value : C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual
        Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\cppwinrt;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\ATLMFC\include;C:\Program Files
        (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um;C:\Program Files
        (x86)\Windows Kits\10\include\10.0.18362.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\cppwinrt;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\ATLMFC\include;C:\Program Files
        (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um;C:\Program Files
        (x86)\Windows Kits\10\include\10.0.18362.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\cppwinrt;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\ATLMFC\include;C:\Program Files
        (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um;C:\Program Files
        (x86)\Windows Kits\10\include\10.0.18362.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared;C:\Program Files (x86)\Windows
        Kits\10\include\10.0.18362.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwin
        rt;;c:\OpenSSL\Win32\include;C:\MySQL\release\include`

What am I missing?!?

**Comment by zosrothko** [comment]
[https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/](https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/)
